### PR TITLE
muontomo-detid-validity-plus-new-features-plus-documentation-plus-cleaning

### PR DESCRIPTION
**Pull request by indra-ehep**
#### PR description:

1. Checks added for detid validity. This has been found to spot the origin of the problem observed during DQM study. See related presentations at,
a) See "2022-10-10_HGCal_in_CMSSW.pdf" by Indranil at https://indico.cern.ch/event/1208397/
b) See Sunada's presentation at https://indico.cern.ch/event/1210093/

2. Added feature to run muon tomography for selected layers as provided in input instead of all layers. 
Expected command will be like,

cmsRun testHGCalCellHitSum_cfg.py layers=1,4,26,41

in case results need to be produced for layer number 1,4,26 and 41.

3. Added feature to mark the first three cell indices(added by Pruthvi). 
4. Now can read wafer input files corresponding to HGCAL geometry version v15, v16, v17. 
5. Added further documentation. 
6. Code cleaning.

#### PR validation:

runTheMatrix test is not applicable since it produces a separate tomography related output file.
Please perform standard test like coding convention and related tests.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Backport not required merging with 12-6-X is sufficient.

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: Validation/HGCalValidation/test/HGCalCellHitSum.cc [hunk 1]
```diff
@@ -1174,6 +1824,7 @@ void HGCalCellHitSum::fillDescriptions(edm::ConfigurationDescriptions &descripti
   desc.add<edm::InputTag>("simhits", edm::InputTag("g4SimHits", "HGCHitsEE"));
   desc.add<std::string>("detector", "HGCalEESensitive");
   desc.add<edm::FileInPath>("geometryFileName", edm::FileInPath("Validation/HGCalValidation/data/wafer_v17.csv"));
+  desc.add<std::string>("layerList", "1");
```
Comment (smuzaffar) [inline Validation/HGCalValidation/test/HGCalCellHitSum.cc:1827]: @indra-ehep , looks like you already have the correct fix here

**Comment by indra-ehep** [comment]
***@***.*** , Actually during the local build using "scram b", I found that
I have to modify two places 1) cfipython directory and 2) inside the as you
have pointed out. So I assumed that I have commit two changes. Now it is
clear. Thanks.*

**Comment by emanueleusai** [comment]
please test

**Comment by indra-ehep** [comment]
@emanueleusai : Sorry about multiple commits, I  lately noted some wrong histogram binning for eta-phi these are now correct. In addition took this opportunity to add "Gen" to a selected list of histogram titles. This will help to distinguish these histograms from that due to Reco.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by indra-ehep** [comment]
Note that this PR is related to Muon Tomography at the generation level the codes to deal with reconstruction level has been committed in PR #39795

**Comment by indra-ehep** [comment]
@cmsdqm Any feedback ?

**Comment by perrotta** [comment]
+1

**Pull request merged**